src/Interpreters/Access/InterpreterCreateRoleQuery.cpp

```diff
@@ -52,6 +58,15 @@ BlockIO InterpreterCreateRoleQuery::execute()
     if (!query.cluster.empty())
         return executeDDLQueryOnCluster(query_ptr, getContext());
 
+    IAccessStorage * storage = &access_control;
+    MultipleAccessStorage::StoragePtr storage_ptr;
+
+    if (!query.storage_name.empty())
+    {
+        storage_ptr = access_control.getStorageByName(query.storage_name);
+        storage = storage_ptr.get();
+    }
```

Comment:
We can support that logic for other types of access entities (users, etc.), it seems it could be useful.